packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -156,22 +156,27 @@ abstract class ButtonStyleButton extends StatefulWidget {
   /// {@macro flutter.material.ButtonStyleButton.iconAlignment}
   final IconAlignment iconAlignment;
 
-  /// Returns a non-null [ButtonStyle] that's based primarily on the [Theme]'s
-  /// [ThemeData.textTheme] and [ThemeData.colorScheme].
+  /// Returns a [ButtonStyle] that's based primarily on the [Theme]'s
+  /// [ThemeData.textTheme] and [ThemeData.colorScheme], but has most values
+  /// filled out (non-null).
   ///
-  /// The returned style can be overridden by the [style] parameter and
-  /// by the style returned by [themeStyleOf]. For example the default
-  /// style of the [TextButton] subclass can be overridden with its
-  /// [TextButton.style] constructor parameter, or with a
-  /// [TextButtonTheme].
+  /// The returned style can be overridden by the [style] parameter and by the
+  /// style returned by [themeStyleOf]. For example the default style of the
```

Comment:
The original text is a little confusing because "the style returned by [themeStyleOf]" sounds like developers are supposed to override themeStyleOf. What we're really saying is that you can override the style with a button-specific theme, like TextButtonTheme or ElevatedButtonTheme.

Reply:
I agree.  I'm not quite sure how to revise it, thouhgh.  How's this?

Reply:
I think it might be better to just leave out the info about [themeStyle]. It's enough to say that you can override a button's style with its style parameter or with a button-specific theme like TextButtonTheme or ElevatedButtonTheme.